source/unix/openssl_platform_init.c

```diff
@@ -555,6 +559,32 @@ static enum aws_libcrypto_version s_resolve_libcrypto_lib(void) {
     return AWS_LIBCRYPTO_NONE;
 }
 
+static void s_validate_libcrypto_linkage(void) {
+    /* NOTE: the choice of stack buffer size is somewhat arbitrary. it's
+     * possible, but unlikely, that libcrypto version strings may exceed this in
+     * the future. we guard against buffer overflow by limiting write size in
+     * snprintf with the size of the buffer itself. if libcrypto version strings
+     * do eventually exceed the chosen size, this runtime check will fail and
+     * will need to be addressed by increasing buffer size.*/
+    char expected_version[32] = { 0 };
```

Comment:
trivial: Why cut it so close? The sample string on the [OpenSSL docs](https://www.openssl.org/docs/man1.0.2/man3/SSLeay_version.html) "OpenSSL 1.0.1a 15 Oct 2015" was 26 chars.
```suggestion
    char expected_version[64] = { 0 };
```